lib/gpt/core/covariant.py

```diff
@@ -84,3 +84,19 @@ def __init__(self, U, params):
                 gpt.pick_checkerboard(cb, _U[mu], self.U[mu])
 
             self.checkerboard[cb] = shift_base(_U, None)
+
+
+def apply_open_boundaries(field):
```

Comment:
This function should not go into core/covariant.py .  Core does not know about QCD or any related jargon.  Why is open boundaries only a concept for >= dimensions and then only goes in the time direction?  The use in the multi-grid is also not elegant and we need to find a better general ansatz.

Reply:
Outside of the Dirac operator, this file is the only place where the boundary phase (0 for open boundaries) gets multiplied into the gauge field. That's why I deemed it a good idea to put the analogue for the fermions in the same file. However, I don't have strong feelings about this. What would be a better place?
To be honest I don't understand the third sentence. Open boundaries as proposed by Lüscher only act in the temporal direction by construction.

Reply:
Open boundaries are a concept far beyond LQCD.  In the gpt/core module, we should not do things specific to LQCD and its jargon.  This is what the qcd module is for.  You can leave this also up to me.